core/driver/sources/src/main/scala/org/virtuslab/ideprobe/ide/intellij/InstalledIntelliJ.scala

```diff
@@ -219,5 +219,26 @@ final class DownloadedIntelliJ(
   override val ideaProperties: Path =
     root.resolve("bin").resolve("idea.properties").write(ideaPropertiesContent)
 
-  override def cleanup(): Unit = root.delete()
+  override def cleanup(): Unit =
+    probePaths.logExport match {
+      case Some(path) =>
+        root
+          .resolve("logs")
+          .copyDir(path.resolve(getPathWithVersionNumber(root).getFileName).resolve("logs"))
```

Comment:
just make this method return the name. Also use .name from Path extensions

Reply:
But in this case I need to use the `copyDir` method, which needs a **Path** not a **String**. Moreover, the `.resolve("logs")` part also must be invoked on an object of type **Path**. That's why the `getPathWithVersionNumber` method should return a **Path**.

Reply:
You are calling `resolve` on `path`. Resolve accepts both `Path` and `String` so nothing should stop you from changing `getPathWithVersionNumber(root).getFileName` to for example: `getIntelliJInstanceIdentifier(root)` - where the new method would look just like old one but with `.name` call before return.

Reply:
ok, done 👍